lib/Doctrine/DBAL/Migrations/Version.php

```diff
@@ -454,16 +453,17 @@ private function formatParamsForOutput(array $params, array $types)
             return '';
         }
 
-        $platform = $this->connection->getDatabasePlatform();
-        $out      = [];
+        $out = [];
         foreach ($params as $key => $value) {
-            $type = $types[$key] ?? 'string';
-            if (Type::hasType($type)) {
-                $outval = Type::getType($type)->convertToDatabaseValue($value, $platform);
-            } else {
-                $outval = '?';
-            }
-            $out[] = is_string($key) ? sprintf(':%s => %s', $key, $outval) : $outval;
+            $type   = $types[$key] ?? 'string';
+            $outval = $this->formatParameter($value, $type);
+            $out[]  = is_string($key) ? sprintf(':%s => %s', $key, $outval) : $outval;
+        }
+
+        if (count($out) > 1) {
+            $out = array_map(function ($element) {
+                return '[' . $element . ']';
```

Comment:
Couldn't we solve this when generating the value (in line 458) instead of re-iterating `$out`? Like:

```diff
- $outval = $this->formatParameter($value, $type);
+ $outval = '[' . $this->formatParameter($value, $type) . ']';
```

Or only when dealing with arrays, in line 497:

```diff
- return implode(', ', array_map([$this, 'parameterToString'], $value));
+ return '[' . implode(', ', array_map([$this, 'parameterToString'], $value)) . ']';
```

The output would be a bit more understandable.